docs/design.md

```diff
@@ -237,6 +238,21 @@ status: # the response
 
 In the future other mechanisms may be defined to provide conventions other than webhooks. In particular, mechanisms that are safe to execute within the controller process like a YTT overlay or WebAssembly. Each mechanism will define the specifics of its own contract similar in scope to the `PodConventionContext`.
 
+#### Webhook Contract 
```

Comment:
Emphasize this section defines one way to implement a webhook, but is not the only way. Another option is to move this content into a `doc.go` file in the webhook package. And use the samples to show how the helper library is actually used.

```suggestion
#### Webhook Helper Library 
```